test/libsolidity/smtCheckerTests/inheritance/base_contract_assertion_fail_8.sol

```diff
@@ -0,0 +1,18 @@
+pragma experimental SMTChecker;
+
+abstract contract A {
+	uint x;
+	function f() public view {
+		assert(x == 2);
+	}
+}
+
+contract C is  A {
+	function g() public {
+		x = 2;
+		f();
+		x = 0;
+	}
+}
+// ----
+// Warning 6328: (94-108): CHC: Assertion violation happens here.\nCounterexample:\nx = 0\n\nTransaction trace:\nA.constructor()\nState: x = 0\nA.f()
```

Comment:
This is wrong, right?

Reply:
Ah right, I see what you mean now. Yea, that's a false positive since the contract cannot be deployed. It's a different problem and I would like to do more checks and add more tests for that though, so I would merge this PR first and do that separately.

Reply:
Created https://github.com/ethereum/solidity/issues/10865 to track this